# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux26mips/hotfix2.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

include/asm/delay.h:74:3: error: impossible constraint in 'asm'

h constraint removed in gcc-4.4 and up

	- Rene Rebe <rene@exactcode.de>

--- linux-2.6.19/include/asm-mips/delay.h.vanilla	2011-02-06 14:26:47.006874950 +0000
+++ linux-2.6.19/include/asm-mips/delay.h	2011-02-06 14:34:19.619804957 +0000
@@ -71,9 +71,10 @@
 		: "r" (usecs), "r" (lpj)
 		: GCC_REG_ACCUM);
 	else if (sizeof(long) == 8)
-		__asm__("dmultu\t%2, %3"
-		: "=h" (usecs), "=l" (lo)
-		: "r" (usecs), "r" (lpj)
+		__asm__("dmultu\t%2, %3 \n\t" 
+			"mfhi\t%0 \n\t"
+		: "=r" (usecs), "=l" (lo) 
+		: "r" (usecs), "r" (lpj) 
 		: GCC_REG_ACCUM);
 
 	__delay(usecs);
